nano/test_common/system.cpp

```diff
@@ -174,19 +194,43 @@ std::shared_ptr<nano::node> nano::test::system::add_node (nano::node_config cons
 	return node;
 }
 
+// TODO: Merge with add_node
 std::shared_ptr<nano::node> nano::test::system::make_disconnected_node (std::optional<nano::node_config> opt_node_config, nano::node_flags flags)
 {
 	nano::node_config node_config = opt_node_config.has_value () ? *opt_node_config : default_config ();
 	auto node = std::make_shared<nano::node> (io_ctx, nano::unique_path (), node_config, work, flags);
-	if (node->init_error ())
+	for (auto i : initialization_blocks)
```

Comment:
make_disconnected_node() is typically used when someone wants a node without any initialisation and no network connections. the name I gave it isn't ideal, in hindsight. It replaced code that was constructing nano::node objects without the use of system resources.

So, I think it would be best to not any additional initialisation.
<img width="1113" alt="image" src="https://github.com/nanocurrency/nano-node/assets/3601511/1d646742-c086-42e6-9075-08f52b1a8d79">

Reply:
By using initialisation_blocks, we are changing the behaviour of many tests and possibly making them pass without testing the intended behaviour.

Reply:
Thanks for pointing this out.  I'll be revisiting the way nodes are added to the system with subsequent io context rework.